### PR TITLE
Add UserTeleportSpawnEvent

### PR DESCRIPTION
**Pull request by Ansandr**
### Details ###
I propose to add an event for listening to the player's teleportation to the spawn, and include it in the new api v2

I've recently had a case myself where having this event would have been a lot easier than having to listen to the `PlayerCommandPreprocessEvent`.

**Proposed feature:**  
Event is called when the user teleports to the spawn using command

**Environments tested:**    

OS: Windows 10 x64

Java version: 16 x64

- [x] Most recent Paper version (1.XX.Y, git-Paper-BUILD)
- [ ] CraftBukkit/Spigot/Paper 1.12.2
- [ ] CraftBukkit 1.8.8

### CODE REVIEW

File: EssentialsSpawn/src/main/java/net/essentialsx/api/v2/events/UserTeleportSpawnEvent.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package net.essentialsx.api.v2.events;
+
+import net.ess3.api.IUser;
+import org.bukkit.Location;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+/**
+ * Called when a user is teleported home via the /spawn command.
```
Comment (JRoy) [inline EssentialsSpawn/src/main/java/net/essentialsx/api/v2/events/UserTeleportSpawnEvent.java]:
```suggestion
 * Called before a user is teleported to spawn via the /spawn command.
```

File: EssentialsSpawn/src/main/java/net/essentialsx/api/v2/events/UserTeleportSpawnEvent.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        this.user = user;
+        this.spawnGroup = spawnGroup;
+        this.target = target;
+    }
+
+    /**
+     * Returns the user who is being teleported
+     *
+     * @return The teleportee.
+     */
```
Comment (JRoy) [inline EssentialsSpawn/src/main/java/net/essentialsx/api/v2/events/UserTeleportSpawnEvent.java]:
```suggestion
    /**
     * @return The user who is being teleported to spawn.
     */
```

File: EssentialsSpawn/src/main/java/net/essentialsx/api/v2/events/UserTeleportSpawnEvent.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+     * @return Name of group
+     */
+    public String getSpawnGroup() {
+        return spawnGroup;
+    }
+
+    /**
+     * Returns the location the user is teleporting to.
+     *
+     * @return Teleportation destination location.
```
Comment (JRoy) [inline EssentialsSpawn/src/main/java/net/essentialsx/api/v2/events/UserTeleportSpawnEvent.java]:
```suggestion
     * The spawn location of the {@link #getUser() user's} {@link #getSpawnGroup() group}.
     * @return The spawn location of the user's group.
```

File: EssentialsSpawn/src/main/java/net/essentialsx/api/v2/events/UserTeleportSpawnEvent.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+     * @return The teleportee.
+     */
+    public IUser getUser() {
+        return user;
+    }
+
+    /**
+     * Return the group of spawn
+     *
+     * @return Name of group
```
Comment (JRoy) [inline EssentialsSpawn/src/main/java/net/essentialsx/api/v2/events/UserTeleportSpawnEvent.java]:
```suggestion
     * The {@link #getUser() user's} group used to determine their spawn location.
     * @return The user's group.
```

**Review by JRoy — CHANGES_REQUESTED** [review]
Few small grammer things, also would be nice for there to be a getController method to get the person who caused the teleport (so if someone did `/spawn PlayerName`), or null if there is none.

**Comment by Ansandr** [comment]
> 
> 
> Few small grammer things, also would be nice for there to be a getController method to get the person who caused the teleport (so if someone did `/spawn PlayerName`), or null if there is none.

Done

**Review by JRoy — APPROVED** [review]

**Comment by DereC4** [comment]
UserTeleportSpawnEvent

**Pull request merged**